projects/step-frontend/src/lib/modules/execution/services/active-execution-context.service.ts

```diff
@@ -1,35 +1,37 @@
 import { inject, Injectable } from '@angular/core';
-import { ActiveExecutionsService } from './active-executions.service';
-import { BehaviorSubject, filter, map, of, shareReplay, switchMap } from 'rxjs';
+import { ActiveExecution, ActiveExecutionsService } from './active-executions.service';
+import { BehaviorSubject, distinctUntilChanged, filter, map, Observable, of, shareReplay, switchMap, tap } from 'rxjs';
 import { takeUntilDestroyed } from '@angular/core/rxjs-interop';
+import { Execution } from '@exense/step-core';
 
 @Injectable()
 export class ActiveExecutionContextService {
-  private _activeExecutions = inject(ActiveExecutionsService);
+  private _activeExecutionsService = inject(ActiveExecutionsService);
 
   private executionIdInternal$ = new BehaviorSubject<string>('');
 
-  readonly executionId$ = this.executionIdInternal$.pipe(
+  readonly executionId$: Observable<string> = this.executionIdInternal$.pipe(
     filter((id) => !!id),
+    distinctUntilChanged((a, b) => a === b),
     takeUntilDestroyed(),
   );
 
-  readonly activeExecution$ = this.executionId$.pipe(
-    map((id) => this._activeExecutions.getActiveExecution(id)),
+  readonly activeExecution$: Observable<ActiveExecution> = this.executionId$.pipe(
+    map((id) => this._activeExecutionsService.getActiveExecution(id)),
     shareReplay(1),
     takeUntilDestroyed(),
   );
 
-  readonly execution$ = this.executionId$.pipe(
-    map((id) => this._activeExecutions.getActiveExecution(id)),
+  readonly execution$: Observable<Execution> = this.executionId$.pipe(
```

Comment:
Typescript will automatically determine the type. Also the result type will be Observable<Execution | undefined>
```suggestion
  readonly execution$ = this.executionId$.pipe(
```